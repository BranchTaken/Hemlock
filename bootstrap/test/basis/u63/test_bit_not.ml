open! Basis.Rudiments
open! Basis
open U63
open Format

let test () =
  printf "@[<h>";
  let rec test = function
    | [] -> ()
    | x :: xs' -> begin
        printf "bit_not %a -> %a\n"
          pp_x x pp_x (bit_not x);
        test xs'
      end
  in
  let xs = [
    0;
    0x7fff_ffff_ffff_ffff
  ] in
  test xs;
  printf "@]"

let _ = test ()
