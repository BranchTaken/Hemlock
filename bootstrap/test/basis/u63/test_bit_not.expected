bit_not 0x0000000000000000 -> 0x7fffffffffffffff
bit_not 0x7fffffffffffffff -> 0x0000000000000000
